### DISCUSSION TITLE
Rewarding apps for depth of integration

### DISCUSSION THREAD

**Discussion by skyf0xx** [kinfoundation/kin-rfcs#178] [KRE 4.0 Discussion]
Proposal here:
https://github.com/kin-labs/rewards-engine/blob/master/improvement-proposals/wg_kre_kin_integration.md

**Comment by mocolicious** [comment]
The main issue I can see with this is that it requires a lot of human intervention and judgement, and the rules seem that they can become arbitrary and could lead to issues or even lawsuits over fair application of these rules.  The current KRE4.0 proposal has a lot of points where this can be a big issue.  A system that operates using SPL programs and purely judges based on the data that can be detected on chain would be less sophisticated but will be less expensive to administer and will always be fair.  The tricky part is finding the right things to detect.

The current Kin memo is a good starting point.  The issue is that it doesn't care as much about Transaction Type as it did in the past.  So many people just leave it out.  We can engineer a more sophisticated application of the memo that can get us to a 99% solution for something that cannot be gamed AND will not be faced with the same issues of arbitrary judgement.

**Reply by skyf0xx** [reply]
How would a memo be able to judge depth of integration?
Perhaps a way to make depth objective though - is to get the average daily spends for active users.

If active users in an app - e.g. PeerBet are spending 6 times, vs Kik (0.01) - PeerBet would have a deeper integration.

**Reply by mocolicious** [reply]
I dont have an exact answer to that yet, gimme some time to simmer on it, but I know we can do better engineering of a solution to this than a simple enum of TransactionType.  It must incentivize rich on-chain data and by design prevent gaming of the labeling of transactions.  The previous iterations seemed that you could just label transactions with whatever and it wouldn't reflect how its actually being used.  For example, labelling a hot wallet transaction as a spend transaction by simply changing the memo.  I agree that daily spends could potentially be a less sophisticated metric that could be used.

It wouldn't be a perfect solution but if we could solve that problem, without requiring human intervention, I think that it could create a strong and fair foundation to build on.  I think the biggest issue of the KRE4 proposed solution is that it takes us further away from the value that blockchain provides.  If we can embrace the technology and also try to solve some of the issues of past iterations, we can move in a positive direction towards a more appropriate and sophisticated solution that would be in the spirit of the original ideals.
